pkg/testing/ig/ig.go

```diff
@@ -0,0 +1,136 @@
+// Copyright 2019-2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ig
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"testing"
+
+	"github.com/blang/semver"
+
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/testing/command"
+)
+
+type IG struct {
+	path    string
+	image   string
+	version semver.Version
+
+	command.Command
+	flags []string
+}
+
+func (ig *IG) createCmd() {
+	args := append([]string{"run", ig.image}, ig.flags...)
+	cmd := exec.Command(ig.path, args...)
+
+	ig.Cmd = cmd
+}
+
+type option func(*IG)
+
+func WithPath(path string) option {
+	return func(ig *IG) {
+		ig.path = path
+	}
+}
+
+func WithPathFromEnvVar() option {
+	return func(ig *IG) {
+		ig.path = os.Getenv("IG")
+	}
+}
+
+func WithImage(image string) option {
+	return func(ig *IG) {
+		ig.image = fmt.Sprintf("%s/%s:%s", os.Getenv("GADGET_REPOSITORY"), image, os.Getenv("GADGET_TAG"))
```

Comment:
IG will already add the default repo and tag if they are not present, don't need to do it here.